From 00d2043626341cf656e597a6e4898a9588d2d84c Mon Sep 17 00:00:00 2001
From: Vignesh Viswanathan <quic_viswanat@quicinc.com>
Date: Thu, 17 Aug 2023 14:53:57 +0530
Subject: [PATCH] Revert "remoteproc: qcom: Trigger BUG_ON from SSR notify
 crash handler"

This reverts commit 1f6a003569cd8d068d56225a5ef840a2ea108cc6.

This change was added as a temporary WAR until the SSR notifier issue
for fatal notification is fixed. Revert this change now as the SSR
notifier is fixed via I263c8d9f6f80b33b9233f8662cbbbe93ea3d7cec and on
receiveing the fatal notification, the client driver will trigger the
BUG_ON.

Change-Id: I92f9734e5bccea8cfab89cd6119c73beff17fdc1
Signed-off-by: Vignesh Viswanathan <quic_viswanat@quicinc.com>
---
 drivers/remoteproc/qcom_common.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/remoteproc/qcom_common.c b/drivers/remoteproc/qcom_common.c
index 1aaeb1419506..24270b9e73dc 100644
--- a/drivers/remoteproc/qcom_common.c
+++ b/drivers/remoteproc/qcom_common.c
@@ -519,8 +519,6 @@ static void ssr_notify_crash(struct rproc_subdev *subdev)
 
 	atomic_notifier_call_chain(&ssr->info->atomic_notifier_list,
 				 QCOM_SSR_NOTIFY_CRASH, &data);
-
-	BUG_ON(1);
 }
 
 /**
-- 
2.34.1

